### PR TITLE
Optimize TextEditorModel's InvalidLineRange from class to struct, add comprehensive test coverage, fix exception handling for TokenizeViewPort

### PR DESCRIPTION
**Pull request by udlose**
#### PR Classification
Code cleanup and test enhancement to improve resource management, thread-safety, and test coverage for `TextEditorModel`.

#### PR Summary
Refactored `TextEditorModel` for safer disposal and better testability, and significantly expanded unit tests for edge cases and internal behaviors.
- `TextEditorModel.cs`: Added thread-safe disposal, converted `InvalidLineRange` to a struct, improved event handler safety, and refactored tokenization logic.
- `TextEditorModelTests.cs`: Added comprehensive tests for disposal, event handling, edge cases, and internal logic
- Change regex instances in LinkElementGenerator to `GeneratedRegex`

## Test Coverage

image

### CODE REVIEW

**Review by danipen — CHANGES_REQUESTED** [review]

File: test/AvaloniaEdit.Tests/TestUtils/ReflectionTestHelper.cs
[the remark this answers is not in the document]
Comment (udlose) [reply test/AvaloniaEdit.Tests/TestUtils/ReflectionTestHelper.cs:1]: Fair enough. I usually try to avoid changing production code for testing even if it means making something visible from private to internal. I wish there was a "PrivatesVisibleTo" lol.  Ok I will change the visibility if you prefer that.

**Comment by danipen** [comment]
@udlose can you handle the review comments generated by copilot? Thanks!

**Comment by udlose** [comment]
yep - was working on them now - Copilot beat me to getting them pushed in time lol. I'm fixing both PR's. Setting back to draft mode until ready

File: src/AvaloniaEdit/Rendering/LinkElementGenerator.cs [hunk 1]
```diff
@@ -21,134 +21,141 @@
...
+    /// This element generator can be easily enabled and configured using the
+    /// <see cref="TextEditorOptions"/>.
+    /// </remarks>
+    public partial class LinkElementGenerator : VisualLineElementGenerator, IBuiltinElementGenerator
+    {
+        [GeneratedRegex(@"\b[\w\d\.\-\+]+\@[\w\d\.\-]+\.[a-z]{2,6}\b")]
+        private static partial Regex DefaultMailGeneratedRegex();
+
+        [GeneratedRegex(@"\b(https?://|ftp://|www\.)[\w\d\._/\-~%@()+:?&=#!]*[\w\d/]")]
+        private static partial Regex DefaultLinkGeneratedRegex();
```
[the remark this answers is not in the document]
Comment (udlose) [reply src/AvaloniaEdit/Rendering/LinkElementGenerator.cs:39]: fixed

File: src/AvaloniaEdit.TextMate/TextEditorModel.cs [hunk 2]
```diff
@@ -174,8 +181,9 @@ void DocumentOnUpdateFinished(object sender, EventArgs e)
```
[the remark this answers is not in the document]
Comment (udlose) [reply src/AvaloniaEdit.TextMate/TextEditorModel.cs]: fixed

**Comment by udlose** [comment]
@danipen done

**Review by danipen — APPROVED** [review]
LGTM
